url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/DeviceGraphicsDisplay/KitRobotGraphicsDisplay.java

```diff
@@ -0,0 +1,577 @@
+
+	int kitMagicX;
+	int kitMagicY;
+
+	int xStep;
+	int yStep;
+
+	AffineTransform trans;
+
+	Client kitRobotClient;
+	ArrayList<KitGraphicsDisplay> kits = new ArrayList<KitGraphicsDisplay>();
+	KitGraphicsDisplay currentKit;
+
+	public KitRobotGraphicsDisplay(Client cli) {
+
+		invisibleCloud = true;
+		kitRobotLocation = Constants.KIT_ROBOT_LOC;
+		kitRobotClient = cli;
+
+		moveToInitialPosition = Command.moveToInspectionStand;
+		moveToFinalPosition = Command.moveToConveyor;
+		position = Position.conveyorPosition;
+		initialJob = false;
+		finalJob = false;
+
+		sendMessage = Message.sendGoodConveyorDoneMessage;
+
+		returnJob = false;
+
+		jobIsDone = true;
+
+		animCount = 70;
+		animateKitRobot = false;
+		reverse = false;
+		seq = 1;
+
+		velocityDivide = Constants.KIT_VELOCITY_DIVIDE;
+
+		kitRobotPositionX = Constants.KIT_ROBOT_LOC.getX()
+				+ kitRobotClient.getOffset();
+		kitRobotPositionY = Constants.KIT_ROBOT_LOC.getY();
+
+		kitMagicX = Constants.KIT_CONVEYOR_LOC.getX();
+		kitMagicY = Constants.KIT_CONVEYOR_LOC.getY();
+
+		kitLeaveLocation = Constants.KIT_LEAVE_LOC;
+		kitEnterLocation = Constants.KIT_CONVEYOR_LOC;
+		inspectionLocation = Constants.INSPECTION_LOC;
+		location1 = Constants.STAND1_LOC;
+		location2 = Constants.STAND2_LOC;
+
+		currentKit = new KitGraphicsDisplay(cli);
+
+	}
+
+	public void setCommands(Command initialCommand, Command finalCommand) {
+		jobIsDone = false;
+		initialJob = true;
+		this.moveToInitialPosition = initialCommand;
+		this.moveToFinalPosition = finalCommand;
+	}
+
+	// begin paths
+	public void InspectionToLocation1() {
+		setCommands(Command.moveToInspectionStand, Command.moveToLocation1);
+		moveToInitialOrFinal();
+	}
+
+	public void InspectionToLocation2() {
+		setCommands(Command.moveToInspectionStand, Command.moveToLocation2);
+		moveToInitialOrFinal();
+	}
+
+	public void InspectionToGoodConveyor() {
+		setCommands(Command.moveToInspectionStand, Command.moveToGoodConveyor);
+		moveToInitialOrFinal();
+	}
+
+	public void ConveyorToLocation1() {
+		setCommands(Command.moveToConveyor, Command.moveToLocation1);
+		moveToInitialOrFinal();
+	}
+
+	public void ConveyorToLocation2() {
+		setCommands(Command.moveToConveyor, Command.moveToLocation2);
+		moveToInitialOrFinal();
+	}
+
+	public void Location1ToInspectionStand() {
+		setCommands(Command.moveToLocation1, Command.moveToInspectionStand);
+		moveToInitialOrFinal();
+	}
+
+	public void Location2ToInspectionStand() {
+		setCommands(Command.moveToLocation2, Command.moveToInspectionStand);
+		moveToInitialOrFinal();
+	}
+
+	public void InspectionStandToGoodConveyor() {
+		setCommands(Command.moveToInspectionStand, Command.moveToGoodConveyor);
+		moveToInitialOrFinal();
+	}
+
+	public void Location1ToLocation2() {
+		setCommands(Command.moveToLocation1, Command.moveToLocation2);
+		moveToInitialOrFinal();
+	}
+
+	// end paths
+
+	/*
+	 * sets the rotation configurations based on the commands
+	 */
+	public void moveToInitialOrFinal() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 29. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.